CONTRIBUTING.rst

```diff
@@ -0,0 +1,103 @@
+====================
+Contributing to CDAP
+====================
+
+Here are instructions to get you started if you want to contribute to CDAP. 
+
+Security Reports
+================
+
+Please *DO NOT* file an issue for security related problems. 
+Please send your reports to `security@cask.co <mailto:security@cask.co>`__.
+
+Creating Issues
+===============
+
+In order to file bugs or new feature requests, please use http://issues.cask.co.
+
+Feature Requests
+================
+
+While proposing a new feature, we look for:
+
+* An issue in http://issues.cask.co that describes the problem and the action that will be taken to fix it
+
+  * Please review existing proposals to make sure the feature proposal doesn't already exist
+
+* Email the `CDAP mailing list <mailto:cdap-dev@googlegroups.com>`__ 
+
+  * Include "Feature: <Name of the feature>" as the subject of the email
+  * Include a link to the issue
+
+Build Environment
+=================
+
+For instructions on setting up your development environment, please see the
+`BUILD.rst <https://github.com/caskco/cdap/blob/develop/BUILD.rst>`__ file.
+
+Contribution Guidelines
+=======================
+
+Creating Issues
+---------------
+An issue should be created at http://issues.cask.co for any bugs or new features before anybody starts working on it. 
+Please take a moment to check that an issue doesn't already exist before creating a new issue. 
+If you see that a bug or an improvement exists already, add a "+1" to the issue to indicate that you have the same
+problem as well. This will help us in prioritization.
+
+Discuss on the Mailing List
+---------------------------
+We recommend that you discuss your plan on the mailing list 
+`cdap-developers@cask.co <mailto: cdap-developers@cask.co>`__
+before starting to code. This gives us a chance to add feedback and help point you in the right direction.
+
+Pull Requests
+-------------
+We love having pull requests and we do our best to review them as quickly as possible. 
+If your pull request is not accepted on the first try, don't be discouraged. 
+If there is a problem with the implementation, you will receive feedback on what to improve.
+
+Conventions
+-----------
+* Fork the repository and make changes on your fork in a feature branch. The branch name should be 
+  "XXXX-something", where "XXXX" is the number of the issue. 
+
+* Submit the code, with unit tests for the changes and documentation, if applicable. Take a look at 
+  the existing tests to get an idea of what's required. 
+  Run the full test suite on your branch before submitting a pull request.
```

Comment:
We can start here, and see how it goes. If people find it doesn't work, or they need help or resources...one problem with committers doing the builds is that it can encourage sloppy pull requests...